pgx-version-updater/src/main.rs

```diff
@@ -0,0 +1,407 @@
+                                dep_item.as_str().unwrap(),
+                                &args.update_version.as_str(),
+                            );
+
+                            deps_table[package] = value(new_version);
+                        }
+                    }
+                }
+            }
+        }
+
+        if args.show_diff {
+            // Call diff command, it provides the easiest way to show context.
+            let mut child = Command::new("diff")
+                .arg(&filepath)
+                .arg("-U")
+                .arg("5")
+                .arg("-")
+                .stdin(Stdio::piped())
+                .stdout(Stdio::piped())
+                .spawn()
+                .expect("Failed to spawn child process");
+
+            let mut stdin = child.stdin.take().expect("Failed to open stdin");
+            let docstring = doc.to_string();
+
+            std::thread::spawn(move || {
+                stdin
+                    .write_all(docstring.as_bytes())
+                    .expect("Failed to write to stdin");
+            });
+
+            let child_output = child.wait_with_output().expect("Failed to read stdout");
+            let mut diff_output = String::new();
+
+            // Using an iterator here, so we need to keep track of certain things for proper output
+            let mut line_ctr = 0;
+            for line in child_output.stdout.lines().skip(2) {
+                let line = line.unwrap();
+
+                match line.chars().nth(0).unwrap() {
+                    '-' => diff_output.push_str(format!("\n            {}", line.red()).as_str()),
+                    '+' => diff_output.push_str(format!("\n            {}", line.green()).as_str()),
+                    _ => diff_output.push_str(format!("\n           {line}").as_str()),
+                }
+
+                line_ctr += 1;
+            }
+
+            // The "diff" command will not print out anything if there is no difference.
+            if line_ctr == 0 {
+                diff_output.push_str(
+                    format!("\n           {}", "* No detectable diff found".dimmed()).as_str(),
+                )
+            } else {
+                diff_output = format!("\n           {}", "* Diff:".dimmed()) + diff_output.as_str();
+            }
+
+            output.push_str(diff_output.as_str());
+        }
+
+        println!("{output}");
+
+        // Write it out!
+        if !args.dry_run {
+            fs::write(filepath, doc.to_string()).expect("Unable to write file");
+        }
+    }
+}
+
+// Always return full path
+fn fullpath(test_path: String) -> PathBuf {
+    let mut path = PathBuf::new();
+    path.push(test_path.clone());
+
+    if path.is_absolute() {
+        path.canonicalize().unwrap()
+    } else {
+        let current_dir = env::current_dir().expect("Could not get current directory!");
+        path.push(current_dir);
+        path.push(test_path.clone());
```

Comment:
Are you sure you want to build the path, then push the current dir on top of the path, then push the path on top again...? Wouldn't that result in duplicating the path in the path? Why not work with the fact that `env::current_dir()` returns a `PathBuf`?

Reply:
🤔